crates/cheatnet/src/execution/gas.rs

```diff
@@ -27,3 +31,17 @@ fn vm_execution_resources_to_resource_mapping(
     map.extend(execution_resources.builtin_instance_counter.clone());
     ResourcesMapping(map)
 }
+
+fn get_total_vm_usage(resources: &ExecutionResources) -> VmExecutionResources {
+    let total_vm_usage = &resources.vm_resources
+        + &(&get_additional_os_resources(
+            &resources.syscall_counter,
+            TransactionType::InvokeFunction,
+        )
+        .unwrap()
+            - OS_RESOURCES
+                .execute_txs_inner()
+                .get(&TransactionType::InvokeFunction)
+                .expect("`OS_RESOURCES` must contain all transaction types."));
+    total_vm_usage.filter_unused_builtins()
```

Comment:
total_vm_usage consists of resources used by vm (`vm_resources`) and additional resources computed from used syscalls (`get_additional_os_resources`).

Unfortunately `get_additional_os_resources` function adds resources used by os (
```
OS_RESOURCES
     .execute_txs_inner()
     .get(&TransactionType::InvokeFunction)
```
) so we have to subtract them then

Reply:
I think this is comment-worthy